llm-service/app/services/models/providers/bedrock.py

```diff
@@ -75,72 +76,115 @@ def list_all_models(
     ) -> list[dict[str, Any]]:
-        credentials = boto3.Session().get_credentials().get_frozen_credentials()
+        if settings.aws_default_region is None:
+            raise ValueError("AWS default region is not set")
+        credentials = boto3.Session().get_credentials()
+        if credentials is None:
+            raise ValueError("AWS credentials not set")
+        credentials = credentials.get_frozen_credentials()
         base_url = (
             f"https://bedrock.{settings.aws_default_region}.amazonaws.com/"
             "foundation-model-availability/"
         )
         models = BedrockModelProvider.list_all_models(modality)
         available_models = []
-        requests = []
+        aws_requests = []
         for model in models:
-            if (
-                "INFERENCE_PROFILE" in model["inferenceTypesSupported"]
-                or "ON_DEMAND" in model["inferenceTypesSupported"]
-            ):
-                model_id = model["modelId"]
-                url = unquote(f"{base_url}{model_id}")
-                request = AWSRequest(method="GET", url=url, headers={})
+            model_id = model["modelId"]
+            url = unquote(f"{base_url}{model_id}")
+            request = AWSRequest(method="GET", url=url, headers={})
 
-                # Sign the request
-                SigV4Auth(credentials, "bedrock", settings.aws_default_region).add_auth(
-                    request
-                )
+            # Sign the request
+            SigV4Auth(credentials, "bedrock", settings.aws_default_region).add_auth(
+                request
+            )
 
-                requests.append((url, dict(request.headers)))
+            aws_requests.append((url, dict(request.headers)))
 
-        responses = []
-        for url, headers in requests:
-            response = requests.get(url, headers=dict(headers))
+        def get_aws_responses(
+            unquoted_url: str, headers: dict[str, str]
+        ) -> dict[str, Any]:
+            """Fetch responses from AWS for the given requests."""
+            response = requests.get(unquoted_url, headers=headers)
             raise_for_http_error(response)
-            responses.append(response.json())
+            return cast(dict[str, Any], response.json())
+
+        responses: list[dict[str, Any] | None] = [None for _ in aws_requests]
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            future_to_index = {
+                executor.submit(get_aws_responses, url, headers): idx
+                for idx, (url, headers) in enumerate(aws_requests)
+            }
+            for future in concurrent.futures.as_completed(future_to_index):
+                idx = future_to_index[future]
+                try:
+                    responses[idx] = future.result()
+                except Exception as e:
+                    print(
+                        f"Error fetching data for model {models[idx]['modelId']}: {e}"
+                    )
```

Comment:
We should try to favor using logging over `print()`

```suggestion
                    logger.exception("Error fetching data for model %s", models[idx]['modelId'])
```